File: src/JavaScript/CreationalPatterns/AbstractFactory/Furniture/concreteFactories/classicFurnitureFactory.js

const IFurnitureFactory = require("../interfaces/furnitureFactory");
const ClassicCabinet = require("../products/classicCabinet");
const ClassicChair = require("../products/classicChair");
const ClassicDiningTable = require("../products/classicDiningTable");

/**
 * @class
 * @implements {IFurnitureFactory}
 */
class ClassicFurnitureFactory {
    /**
     * @override
     */
    createCabinet() {
        return new ClassicCabinet();
    }

    /**
     * @override
     */
    createChair() {
        return new ClassicChair();
    }

    /**
     * @override
     */
    createDiningTable() {
        return new ClassicDiningTable();
    }
}

module.exports = ClassicFurnitureFactory;

File: src/C#/CreationalPatterns/AbstractFactory/Furniture/ConcreteFactories/ClassicFurnitureFactory.cs
using Furniture.Interfaces;
using Furniture.Products;
using System;
using System.Collections.Generic;
using System.Text;

namespace Furniture.ConcreteFactories
{
    public class ClassicFurnitureFactory : IFurnitureFactory
    {
        public IFurniture CreateCabinet()
        {
            return new ClassicCabinet();
        }

        public IFurniture CreateChair()
        {
            return new ClassicChair();
        }

        public IFurniture CreateDiningTable()
        {
            return new ClassicDiningTable();
        }
    }
}
